.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,23 +1,17 @@
-Any of the checked boxes below indicate that I took action:
+## Context
+<!--- Why do you believe many users will benefit from this change? -->
+<!--- Link to relevant issues or forum discussions here -->
 
-- [ ] Reviewed the [Contribution Guidelines](https://github.com/gradle/gradle/blob/master/.github/CONTRIBUTING.md).
-- [ ] Signed the [Gradle CLA](http://gradle.org/contributor-license-agreement/).
-- [ ] Ensured that basic checks pass: `./gradlew quickCheck`
+#### Contributor Checklist
+- [ ] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/.github/CONTRIBUTING.md)
+- [ ] [Sign Gradle CLA](http://gradle.org/contributor-license-agreement/)
+- [ ] [Link to Design Spec](https://github.com/gradle/gradle/tree/master/design-docs) for changes that affect more than 1 public API or change > 20 files
```

Comment:
Maybe explain what a public API is? I don't think it's necessarily common knowledge. Do we have a page we could link to?